api/v1beta1/metal3cluster_webhook.go

```diff
@@ -68,6 +74,79 @@ func (c *Metal3Cluster) validate() error {
 		)
 	}
 
+	if c.Spec.CloudProviderEnabled != nil && c.Spec.NoCloudProvider != nil {
+		if !*c.Spec.CloudProviderEnabled && !*c.Spec.NoCloudProvider {
```

Comment:
`if !*c.Spec.CloudProviderEnabled && !*c.Spec.NoCloudProvider` and `if *c.Spec.CloudProviderEnabled && *c.Spec.NoCloudProvider` can be combined to  `if *c.Spec.CloudProviderEnabled == *c.Spec.NoCloudProvider`.

Reply:
done